tracer/src/Datadog.Trace/Ci/TestOptimizationTestManagementFeature.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="TestOptimizationTestManagementFeature.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System.Threading.Tasks;
+using Datadog.Trace.Ci.Configuration;
+using Datadog.Trace.Ci.Net;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Ci;
+
+internal class TestOptimizationTestManagementFeature : ITestOptimizationTestManagementFeature
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(TestOptimizationTestManagementFeature));
+    private readonly bool _enabled;
+    private readonly Task<TestOptimizationClient.TestManagementResponse> _testManagementTask;
+
+    private TestOptimizationTestManagementFeature(TestOptimizationSettings settings, TestOptimizationClient.SettingsResponse clientSettingsResponse, ITestOptimizationClient testOptimizationClient)
+    {
+        if (settings is null)
+        {
+            ThrowHelper.ThrowArgumentNullException(nameof(settings));
+        }
+
+        if (testOptimizationClient is null)
+        {
+            ThrowHelper.ThrowArgumentNullException(nameof(testOptimizationClient));
+        }
+
+        if (settings.TestManagementEnabled == true || clientSettingsResponse.TestManagement.Enabled == true)
+        {
+            Log.Information("TestOptimizationTestManagementFeature: Test management is enabled.");
+            settings.SetTestManagementEnabled(true);
+            _testManagementTask = Task.Run(() => InternalGetTestManagementTestsAsync(testOptimizationClient));
```

Comment:
Just wondering, is there value to using `Task.Run()` here over calling them directly 🤔  e.g.

```suggestion
            _testManagementTask = InternalGetTestManagementTestsAsync(testOptimizationClient);
```

Reply:
I do that because some of the http transaction requires Git information and the Git process commands are sync, And I just want to complete the `.ctor` the fastest as possible, any wait for the resolution of these task can be done later.

Reply:
Yeah, but in neither case are we `await`-ing the task, so just wondering if wrapping it in a `Task.Run` is necessary? 🤔